benchmark/engine.py

```diff
@@ -0,0 +1,324 @@
+import json
+import os
+import sys
+import time
+import uuid
+from enum import Enum
+
+import boto3
+import dialogflow_v2 as dialogflow
+import soundfile
+
+
+def _path(x):
+    return os.path.join(os.path.dirname(__file__), '../%s' % x)
+
+sys.path.append(_path('rhino/binding/python'))
+from rhino import Rhino
+
+
+class NLUEngines(Enum):
+    AMAZON_LEX = 'AMAZON_LEX'
+    DIALOGFLOW = 'DIALOGFLOW'
+    PICOVOICE_RHINO = 'PICOVOICE_RHINO'
+
+
+class NLUEngine(object):
+    def process_file(self, path):
+        raise NotImplementedError()
+
+    def process(self, folder):
+        raise NotImplementedError()
+
+    def __str__(self):
+        raise NotImplementedError()
+
+    @classmethod
+    def create(cls, engine_type):
+        if engine_type is NLUEngines.AMAZON_LEX:
+            return AmazonLex()
+        elif engine_type is NLUEngines.DIALOGFLOW:
+            return Dialogflow()
+        elif engine_type is NLUEngines.PICOVOICE_RHINO:
+            return PicovoiceRhino()
+        else:
+            raise ValueError("cannot create %s of type '%s'" % (cls.__name__, engine_type))
+
+
+class AmazonLex(NLUEngine):
+    def __init__(self):
+        self._client = boto3.client('lex-runtime')
+
+    def process_file(self, path):
+        cache_path = path.replace('.wav', '.amazonlex')

```

Comment:
lets make sure we add all the cache files to .gitignore so that we dont commit them to github.